File: samples/syntax/sample.php
<?php

final class Point
{
    public function __construct(
        private int $x,
        private int $y,
    ) {}

    public function norm2(): int
    {
        return $this->x ** 2 + $this->y ** 2;
    }
}

echo (new Point(3, 4))->norm2(), PHP_EOL;

File: samples/syntax/sample.lua
local Point = {}
Point.__index = Point

function Point.new(x, y)
  return setmetatable({ x = x, y = y }, Point)
end

function Point:norm2()
  return self.x * self.x + self.y * self.y
end

print(Point.new(3, 4):norm2())
